metagraph/src/common/vectors/vector_algorithm.hpp

```diff
@@ -489,6 +676,49 @@ prev_bit(const t_int_vec &v,
 }
 
 
+// Return an int_vector whose limbs are aligned to alignment bytes.
+// This is useful for algorithms requiring access to larger words in the underlying
+// vector (atomic increment, SIMD, etc.)
+template <int width = 0>
+inline sdsl::int_vector<width> aligned_int_vector(size_t size = 0, uint64_t val = 0,
+                                                  uint8_t var_width = 0,
+                                                  size_t alignment = 8) {
+    assert(__builtin_popcountll(alignment) == 1);
+
+    // This is a dirty hack to allow for reallocating an int_vector<width>'s
+    // underlying storage
+    struct int_vector_access {
+        typename sdsl::int_vector<width>::size_type m_size;
+        uint64_t *m_data;
+        typename sdsl::int_vector<width>::int_width_type m_width;
+    };
+    static_assert(sizeof(sdsl::int_vector<width>) == sizeof(int_vector_access));
+    assert(!width || width == var_width);
+
+    sdsl::int_vector<width> v;
+    auto &v_cast = reinterpret_cast<int_vector_access&>(v);
+    v_cast.m_size = size * var_width;
+    v_cast.m_width = var_width;
+    free(v_cast.m_data);
+
+    size_t capacity_bytes = (((((v_cast.m_size + 63) >> 6) << 3) + alignment - 1) / alignment) * alignment;
+    if (posix_memalign((void**)&v_cast.m_data, alignment, capacity_bytes) || !v_cast.m_data)
+        throw std::bad_alloc();
+
+    memset(v_cast.m_data, 0, capacity_bytes);
+
+    if (val)
+        sdsl::util::set_to_value(v, val);
+
+    return v;
+}
+
+inline sdsl::bit_vector aligned_bit_vector(size_t size = 0, bool val = false,
```

Comment:
```suggestion
inline sdsl::bit_vector aligned_bit_vector(size_t size = 0, bool val = 0,
```

Reply:
Any particular reason why?

Reply:
Minor point. I think it's just easier to read when it is in this context -- initialize a vector, even though it's binary. Otherwise feels like this may be another flag like `async`, or something...